### ISSUE TITLE
If music is enabled on build engine games they hang the computer

### ISSUE THREAD

**Issue by Nemo1985** [crazii/SBEMU#149]
System specs
Motherboard: ECS [P4S5A/DX+(V5.1A)] (https://www.ecs.com.tw/he/Product/Motherboard/P4S5A-DX_Plus/overview)
Audio C-Media CMI9738 4-channel audio CODEC Compliant with AC'97 2.2 specification
Cpu: Pentium 4 3.06ghz HT disabled
Memory: 2gb of ddr 400 (running at 333)
Storage: SSD 64gb.
OS: Windows 98 SE (as dos 7.10)

Build games (I tested duke nukem 3d and blood both shareware version) hang the system if the music is enabled in setup (as sound blaster or anything which produce an output), while if only the sound is enabled everything works fine. 
It happens both in real mode or protected mode (jemmex or emm386). I tried different switches for sbemu without any different result (I've also tried to rem everything except jemmex and sbemu just to avoid any possible incompatibility without any different outcome, I tried to change the irq to 5, use the 22050 hertz value, disable fixtc, with or without /t switch).

Blood crashes usually in few seconds while Duke Nukem needs more time. I also noticed that during the loading of duke nukem 3d the screen gets messed up for a brief moment when music is enabled.
Only once I was able to get a error screen (before the system hanged completely as always):
![photo_2024-10-28_02-04-42](https://github.com/user-attachments/assets/8b81acdc-357c-4d93-8194-51b4d6f5ca08)

I also get a crash of sbemu if I use the /SCFM or /SCMPU switch:
![photo_2024-10-26_11-45-27](https://github.com/user-attachments/assets/ccf58982-ae9f-4859-9321-4f411305c2b2)

Config.sys

SWITCHES=/E:288 /F
DOS=HIGH,UMB,NOAUTO
BUFFERSHIGH=30,0
FILESHIGH=60
FCBSHIGH=4,0
STACKSHIGH=9,256
LASTDRIVEHIGH=Z
COUNTRY=039,850,C:\WINDOWS\COMMAND\COUNTRY.SYS
DEVICE=C:\UTIL\JEMMEX.EXE NOEMS X=TEST I=TEST I=B000-B7FF
DEVICEHIGH=C:\WINDOWS\COMMAND\DISPLAY.SYS CON=(EGA,,1)
DEVICEHIGH=C:\UTIL\XHDD.SYS /400 /H /O /P
DEVICEHIGH=C:\UTIL\XDVD2.SYS /D:OPTICAL /H

Autoexec.bat
@ECHO OFF
PROMPT $p$g
PATH C:\WINDOWS;C:\UTIL
SET TEMP=C:\WINDOWS\TEMP
MODE CON CODEPAGE PREPARE=((850) C:\WINDOWS\COMMAND\EGA.CPI)
MODE CON CODEPAGE SELECT=850
rem LH C:\UTIL\MOUSE.EXE
LH KEYB IT,,C:\WINDOWS\COMMAND\KEYBOARD.SYS
LH C:\UTIL\DOSKEY.COM
LH C:\UTIL\CTMOUSE.EXE /R2 /O
C:\UTIL\SHSUCDX.COM /D:OPTICAL
C:\UTIL\SBEMU\JLOAD.EXE C:\UTIL\SBEMU\QPIEMU.DLL
C:\UTIL\SBEMU\HDPMI32I.EXE -R -X
LH C:\UTIL\SBEMU\SBEMU.EXE /FIXTC /K44100 /T6
LH C:\UTIL\FDAPM.COM /ADV:REG
C:\UTIL\MTRRLFBE.EXE LFB WC

SBEMU /SLC:

QEMM or QPIEMU not installed, disabling real mode support.
Autodetecting/testing available outputs/soundcards, please wait...
1: ICH : Intel SI7012 found on port:D800 irq:10 (type:SIS7012, bits:16)
Autodetecting finished... Exiting...

In a last desperate trying I downloaded the image, burned to a usb drive and booted from it, same result, blood hangs in less than 10 seconds.

**Comment by Nemo1985** [comment]
There is noone taking care of troubleshooting?
Great

**Comment by volkertb** [comment]
Judging from multiple reports, there seem to have been some regressions in recent versions of SBEMU.

I know it's over a year and a half old, but could you maybe test with version [1.0beta3](https://github.com/crazii/SBEMU/releases/tag/1.0beta3), so we have a bit of a baseline, and we can rule out whether this also is a regression, or this never worked in the first place?

No rush, though! Whatever time you have and are willing to spare to help test SBEMU is appreciated. Thanks! 🙏🏾

**Comment by Nemo1985** [comment]
> Judging from multiple reports, there seem to have been some regressions in recent versions of SBEMU.
> 
> I know it's over a year and a half old, but could you maybe test with version [1.0beta3](https://github.com/crazii/SBEMU/releases/tag/1.0beta3), so we have a bit of a baseline, and we can rule out whether this also is a regression, or this never worked in the first place?
> 
> No rush, though! Whatever time you have and are willing to spare to help test SBEMU is appreciated. Thanks! 🙏🏾

Thank you for the help, I will try and report the result very soon.

**Comment by Nemo1985** [comment]
> Judging from multiple reports, there seem to have been some regressions in recent versions of SBEMU.
> 
> I know it's over a year and a half old, but could you maybe test with version [1.0beta3](https://github.com/crazii/SBEMU/releases/tag/1.0beta3), so we have a bit of a baseline, and we can rule out whether this also is a regression, or this never worked in the first place?
> 
> No rush, though! Whatever time you have and are willing to spare to help test SBEMU is appreciated. Thanks! 🙏🏾

I tested the version you adviced but it doesn't recognize my sound card so I actually couldn't do any test with games.
Here is the output:
No soundcard found!
Unkown soundcard (output module) name: NULL

**Comment by Nemo1985** [comment]
I suppose I won't hold my breath for a quick bugfix since the developer seems away since last spring

**Comment by volkertb** [comment]
@Nemo1985 I agree that @crazii's absence makes it less likely that issues will be fixed in the short term. In fact, I'm still hoping on more skilled volunteers to join and pick up further development of SBEMU. And I completely understand your skepticism, but I don't think we should have to close this bug report because of that. I'll keep this open for the time being. If you object to this for any reason, please let me know.

Thanks for your efforts to test and report this, and please don't give up on SBEMU. Also, Happy New Year! :slightly_smiling_face:

**Comment by Nemo1985** [comment]
I had the chance to test the beta4, this version also detects my sound card but it has the same issue of the beta5.
To summarize:
Beta 3 my soundcard is not detected
Beta 4 and 5 it works fine except for build games.
